staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -182,6 +185,20 @@ func (d *DiscoveryClient) ServerGroups() (apiGroupList *metav1.APIGroupList, err
 	if len(v.Versions) != 0 {
 		apiGroupList.Groups = append([]metav1.APIGroup{apiGroup}, apiGroupList.Groups...)
 	}
+
+	// populate the map of hashes for each group version
+	d.hashesMu.Lock()
+	defer d.hashesMu.Unlock()
+	d.hashesByGroupVersion = make(map[string]string)
+	for _, apiGroup := range apiGroupList.Groups {
+		if apiGroup.Hash != "" {
+			for _, version := range apiGroup.Versions {
+				groupVersion := schema.GroupVersion{Group: apiGroup.Name, Version: version.Version}
+				d.hashesByGroupVersion[groupVersion.String()] = apiGroup.Hash
```

Comment:
@Jefftree That makes me wonder, don't we want a `Hash` for each `GroupVersion` rather than for each `Group`?
That code looks weird, you're grabbing the hash from the group, but then assign it to the groupVersion, shouldn't the hash be in the `schema.GroupVersion` struct then?

Reply:
Yeah it should be for each GroupVersion. The Group contains a list of GroupVersions, and each GroupVersion should have their own hash

Reply:
Added the hash to the `GroupVersionForDiscovery` struct since this seems be used more like the `APIGroup` (and `GroupVersionForDiscovery` explicitly mentions that it can be extended). Let me know